### PR TITLE
cardano-services CLI refactor + StakePool OpenAPI metrics addition

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
We're about to introduce two new services to the `cardano-services` package to support the custom data projections, which have common dependencies, but the existing program structure doesn't support sharing of common CLI options. It's also carrying some technical debt we are best to pay off now, with a move to better modularity as a replacement.

# Proposed Solution
See commit messages. I scoped the commits purposely to ensure code movement and specific changes are separate, which will help with the review.
 
# Important Changes Introduced
- I've also picked some improvement commits from #609, https://github.com/input-output-hk/cardano-js-sdk/pull/612/commits/805c672685a590b0d05da2117de8257290fd0ba1, https://github.com/input-output-hk/cardano-js-sdk/pull/612/commits/81f10742f2fe24cfe6a2d71ac6f355396fb20755, and  https://github.com/input-output-hk/cardano-js-sdk/pull/612/commits/cb8143a3704b4df086d156b323fbbd00d1e8f12f, since it's on-hold given it's not achieving the desired outcome, at least not entirely.
- This refactor hoists the API URL and build info concepts to the service level, so we can expose a HTTP server for each service, for the purpose of monitoring. It does not add the implementation to the RabbitMQ worker.

### CODE REVIEW

File: packages/cardano-services/src/StakePool/openApi.json [hunk 1]
```diff
@@ -347,6 +348,79 @@
...
+          "livePledge",
+          "saturation",
+          "size",
+          "stake"
+        ],
+        "properties": {
+          "apy": {
+            "anyOf": [
+              {
+                "$ref": "#/components/schemas/Undefined"
```
Comment (rhyslbw) [inline packages/cardano-services/src/StakePool/openApi.json:366]: I've made a ticket to remove this awkward API: `ADP-2735`

File: packages/cardano-services/src/cli.ts [hunk 2]
```diff
@@ -56,75 +51,21 @@ const stringToBoolean = (value: string, program: Programs, option: string) => {
...
+  withOgmiosOptions(
+    withPostgresOptions(
+      withRabbitMqOptions(
+        program
+          .command('start-server')
+          .description('Start the HTTP server')
+          .argument('[serviceNames...]', `List of services to attach: ${Object.values(ServiceNames).toString()}`)
+      )
+    )
+  )
```
Comment (IvayloAndonov) [inline packages/cardano-services/src/cli.ts]: 👍🏼

**Review by IvayloAndonov — APPROVED** [review]
Nice refactoring!🎉 
Just the E2E job didn't pass `txChainHistory.test.ts`, not should if it's related to these changes though. Maybe rerun and check why was failed afterward.

**Comment by rhyslbw** [comment]
Thanks @IvayloAndonov. Looks like a flaky test. Rerunning now

**Review by iccicci — APPROVED** [review]
Very nice set of refactoring! 🚀

**Review by mkazlauskas — APPROVED** [review]
Great work! :rocket:

**Pull request merged**